src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -278,6 +282,23 @@ public String getQuestionResultStatisticsCsv(
     }
 
     /**
+     * Generates a key, value mapping. Each key corresponds to a recipient and its value is the normalised self rank.
+     * @param responses  a list of responses
+     */
+    private Map<String, Integer> generateSelfRankForEachRecipient(List<FeedbackResponseAttributes> responses) {
+        Map<FeedbackResponseAttributes, Integer> normalisedRankOfResponse = getNormalisedRankForEachResponse(responses);
```

Comment:
@VamsiSangam I believe the current approach might be better then, since that would make the code unnecessarily complicated.